central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -344,16 +353,11 @@ func (m *managerImpl) handleReadyScan() {
 				concurrency.WithLock(&m.watchingScansLock, func() {
 					delete(m.watchingScans, scanWatcherResult.WatcherID)
 				})
-				// At the moment we simply do not start the ScanConfigWatcher if there are errors in the ScanWatchers.
-				// There are many reasons why a ScanWatcher might fail like, for example, the Scan was deleted mid-execution.
-				// If this happens, we will generate many ReportSnapshots with timeouts. Until we implement a way to
-				// distinguish legitimate failures (CO not reporting back), we log the error not create the Report.
-				if scanWatcherResult.Error != nil {
-					log.Errorf("The scanResults returned with an error: %v", scanWatcherResult.Error)
-					continue
-				}
```

Comment:
Why we don't want to skip on errors anymore here?

Reply:
This was meant to be removed. The problem is that we have some scans that were reported double one successfully and one with failure and at that moment I didn't know why. Now I know that CO sends sometimes updates for the same scan execution with different timestamps which leads to multiple `scanWatchers` being started but only one finishes successfully. This PR tries to address this by making sure that if we receive a scan error with a timestamp older than the last snapshot then we ignore it:
- New scan is executed
- CO sends an update with timestamp X
- ScanWatcher starts for the timestamp X
- CO sends an update with timestamp Y
- ScanWatcher starts for the timestamp Y
- CO sends the CheckResults with time Y
- ScanWatcher for timestamp Y ends successfully
- ACS generates the report
- The ScanWatcher for timestamp X times out
- ACS ignores this error because we already have a scan reported with timestamp > X

There's still a situation where this is not handled: 

- New scan is executed
- CO sends an update with timestamp X
- ScanWatcher starts for the timestamp X
- CO sends an update with timestamp Y
- ScanWatcher starts for the timestamp Y
- CO sends the CheckResults with no timestamp (this happens in CO versions < 1.6)
- ScanWatcher for timestamp X times out
- ACS does not ignore the error because we don't have a newer scan reported
- ScanWatcher for timestamp Y times out
- ACS does not ignore the error because we don't have a newer scan reported
So in this last situation we have two failed reports when in reality only one scan was executed.

Reply:
Understood! Thanks for adding those examples, that really helps.